src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpTransport.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\AmqpExt;
+
+use Symfony\Component\Messenger\Transport\Serialization\DecoderInterface;
+use Symfony\Component\Messenger\Transport\Serialization\EncoderInterface;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class AmqpTransport implements TransportInterface
+{
+    private $encoder;
+    private $decoder;
+    private $dsn;
+    private $options;
+    private $debug;
+    private $connection;
+    private $receiver;
+    private $sender;
+
+    public function __construct(EncoderInterface $encoder, DecoderInterface $decoder, string $dsn, array $options, bool $debug)
+    {
+        $this->encoder = $encoder;
+        $this->decoder = $decoder;
+        $this->dsn = $dsn;
+        $this->options = $options;
+        $this->debug = $debug;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(callable $handler): void
+    {
+        ($this->receiver ?? $this->getReceiver())->receive($hander);
```

Comment:
Not sure what you mean, doesn't feel like to me :)

Reply:
I think @yceruto mean next variant:

```php
$this->receiver ? $this->receiver->receive($hander) ? $this->getReceiver()->receive($hander)
```

Reply:
I meant `($this->receiver ?: $this->getReceiver())` instead, as `$this->receiver` is always defined I don't see the need for `??`, but probably I'm missing something here :)

Reply:
`$this->` is `null` until `$this->getReceiver()` is called a first time. So make sense to use the **null**-coalescing operator rather than ternary for this :)

Reply:
According to [the manual](http://php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op):
> The null coalescing operator (??) has been added as syntactic sugar for the common case of needing to use a ternary in conjunction with isset(). It returns its first operand if **IT EXISTS** and is not NULL; otherwise it returns its second operand.

So ternary operator fits better in this case as `$this->receiver` always exists, but NVM it's just a detail, `??` works fine too.